app/src/main/java/org/systers/mentorship/view/activities/MemberProfileActivity.kt

```diff
@@ -31,6 +34,21 @@ class MemberProfileActivity : BaseActivity() {
         supportActionBar?.title = getString(R.string.member_profile)
         supportActionBar?.setDisplayHomeAsUpEnabled(true)
 
+
+        profileViewModel= ViewModelProviders.of(this).get(ProfileViewModel::class.java)
```

Comment:
So each time we fetch the other members profile, we also fetch the current logged in user profile? Is there a better way of doing this 🤔 Should this validation be done only when we get to the SendRequest screen and made by the backend endpoint, that checks this logic and responds 🤔 @mtreacy002 what do you think?

Reply:
Is this network costly? if we did this on the backend, we wouldn't use so extra network resources for this 🤔

Reply:
@isabelcosta, I think it comes down to who we want to pay for the process overhead, our users or us. Pushing it to backend means we pay on computing overhead, leaving it at frontend means users pay for the computing overhead 😂. I'm not sure who pays for network overhead, pretty sure both have to cover half-half.
 If we decide to pay for computing overhead, we could add a compare_users method at backend to check mentor/mentee need_mentoring/available_to_mentor status to make sure they're not on the same availability status. Although, this might also impacted on extra processing time on UI when user push the send request button coz we need to do backend computing first. It might appear as lagging (but depends I think on the device itself. Newer phone obviously have better processing power than older ones).
This is just my thoughts. I'll go with however you decide... 😉

Reply:
The caveat is when developers do testing on swagger ui, the input they insert to the fields have to be the valid one coz otherwise wrong data will cause problems.
My suggestion is we keep validation_utils for when .env = local or dev, to validate data input by developers using swagger ui but when .env is production, we use the frontend validations only.

Reply:
I agree that more validation of the input showed be done in the front end @mtreacy002 
I think we may need issues on the frontend for that.

Also we may have to document somewhere the validation rules of the fields, so that we can keep validation in the backend and frontend 🤔

Reply:
@isabelcosta,  I'll open an issue on frontend for this validation then (will check first if there's an open issue on this already or not, if there is, will add link to this discussion there). Also, in regard to 

> Also we may have to document somewhere the validation rules of the fields, so that we can keep validation in the backend and frontend 🤔

I'll open a Zulip discussion on this coz this topic goes beyond this PR 😉.